tests/buttons.py

```diff
@@ -6,167 +6,237 @@
     ofs = step // 2
     return cell * step + ofs
 
 
-def grid35(x: int, y: int, layout_type: LayoutType) -> Coords:
-    return grid(display_width(layout_type), 3, x), grid(
-        display_height(layout_type), 5, y
-    )
+class ScreenButtons:
+    def __init__(self, layout_type: LayoutType):
+        self.layout_type = layout_type
+        self.width = _width(layout_type)
+        self.height = _height(layout_type)
+
+    # 3 columns, 4 rows, 1st row is input area
+    def _grid35(self, x: int, y: int) -> Coords:
+        return _grid(self.width, 3, x), _grid(self.height, 5, y)
+
+    # TODO: do not expose this
+    # 3 columns, 3 rows, 1st row is input area
+    def grid34(self, x: int, y: int) -> Coords:
+        return _grid(self.width, 3, x), _grid(self.height, 4, y)
+
+    # Horizontal coordinates
+    def _left(self) -> int:
+        return _grid(self.width, 3, 0)
+
+    def _mid(self) -> int:
+        return _grid(self.width, 3, 1)
+
+    def _right(self) -> int:
+        return _grid(self.width, 3, 2)
+
+    # Vertical coordinates
+    def _top(self):
+        return _grid(self.height, 6, 0)
+
+    def _bottom(self):
+        return _grid(self.height, 6, 5)
+
+    # Buttons
+
+    # Right bottom
+    def ok(self) -> Coords:
+        return (self._right(), self._bottom())
+
+    # Left bottom
+    def cancel(self) -> Coords:
+        return (self._left(), self._bottom())
+
+    # Mid bottom
+    def info(self) -> Coords:
+        return (self._mid(), self._bottom())
+
+    # Menu/close menu button
+    def menu(self) -> Coords:
+        if self.layout_type in (LayoutType.Bolt, LayoutType.Delizia):
+            return (215, 25)
+        else:
+            raise ValueError("Wrong layout type")
+
+    # Center of the screen
+    def tap_to_confirm(self) -> Coords:
+        assert self.layout_type is LayoutType.Delizia
+        return (_grid(self.width, 1, 0), _grid(self.width, 1, 0))
+
+    # Yes/No decision component
+    def ui_yes(self) -> Coords:
+        assert self.layout_type is LayoutType.Delizia
+        return self.grid34(2, 2)
+
+    def ui_no(self) -> Coords:
+        assert self.layout_type is LayoutType.Delizia
+        return self.grid34(0, 2)
+
+    # +/- buttons in number input component
+    def number_input_minus(self) -> Coords:
+        if self.layout_type is LayoutType.Bolt:
+            return (self._left(), _grid(self.height, 5, 1))
+        elif self.layout_type is LayoutType.Delizia:
+            return (self._left(), _grid(self.height, 5, 3))
+        else:
+            raise ValueError("Wrong layout type")
+
+    def number_input_plus(self) -> Coords:
+        if self.layout_type is LayoutType.Bolt:
+            return (self._right(), _grid(self.height, 5, 1))
+        elif self.layout_type is LayoutType.Delizia:
+            return (self._right(), _grid(self.height, 5, 3))
+        else:
+            raise ValueError("Wrong layout type")
+
+    def word_count_all_word(self, word_count) -> Coords:
```

Comment:
nit: `coords_map.get(word_count)` may return `None` if `word_count` is not in the map.

Reply:
Sorry, I missed the `assert`.